lib/client/rpc/proposals.ts

```diff
@@ -242,7 +251,184 @@ export class FutarchyRPCProposalsClient implements FutarchyProposalsClient {
         ),
       ])
       .instruction();
-
     return ix;
   }
+  createMemoInstruction() {
+    return {
+      programId: new PublicKey("MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr"),
+      accounts: [],
+      data: Buffer.from("Memo", "utf8")
+    }
+  }
+
+  async simulate(tx: Transaction) {
+    const { blockhash } = await this.rpcProvider.connection.getLatestBlockhash();
+    tx.recentBlockhash = blockhash;
+    tx.feePayer = this.rpcProvider.wallet.publicKey; // Assuming the provider has a wallet
+
+    try {
+
+      const simulationResult = this.rpcProvider.connection.simulateTransaction(tx)
+      console.log(simulationResult)
+    } catch (e) {
+      console.error("Simulation error:", e)
+    }
+    // const wireTransaction = tx.
+  }
+
+  async getUserTokenBalance(tokenMintAddress: PublicKey) {
+    // Get the associated token account for the user's wallet and the token mint
+    const userTokenAccountAddress = await getAssociatedTokenAddress(
```

Comment:
I suggest using https://github.com/solana-labs/solana-program-library/blob/d72289c79a04411c69a8bf1054f7156b6196f9b3/token/js/src/state/mint.ts#L190 in place of `getAssociatedTokenAddress` just given that you don't need async for it..